File: python/chapter05/completions.py
import requests
import os
import json

apiKey = os.environ.get("OPENAI_API_KEY")

headers = {
    'Content-Type': 'application/json',
    'Authorization': 'Bearer ' + apiKey
}

data = json.dumps({
    "prompt": "Once upon a time",
    "max_tokens": 15
})

url = 'https://api.openai.com/v1/engines/davinci/completions'

result = requests.post(url, headers=headers, data=data)

# print(result.json()["choices"][0]["text"])

print(json.loads(data)["prompt"] + result.json()["choices"][0]["text"])

File: nodejs/chapter05/completions.js
const axios = require('axios');

const client = axios.create({
  headers: {
    'Authorization': 'Bearer ' + process.env.OPENAI_API_KEY
  }
});

const completionParmas = {
  "prompt": "Once upon a time",
  "max_tokens": 10
}

client.post('https://api.openai.com/v1/engines/davinci/completions', completionParmas)
  .then(result => {
    console.log(completionParmas.prompt + result.data.choices[0].text);
    // console.log(result.data);
  }).catch(err => {
    console.log(err);
  });